Added documentation for point in time recovery (#478)
vitessio/website: content/en/docs/reference/features/recovery.md

---
title: Point In Time Recovery
aliases: ['/docs/recovery/pitr','/docs/reference/pitr/']
---

## Point in time recovery

### Introduction

> This enables to recover data in vitess to a past point in time. There can be multiple recovery request active at same time. It is possible to recover across sharding action i.e. you should be able to recover to a time when there were two shards even though at present there are four.

Reviewer: ```suggestion
The Point in Time Recovery feature in Vitess enables recovery of data to a past point in time. There can be multiple recovery requests active at the same time. It is possible to recover across sharding actions, i.e. you can recover to a time when there were two shards even though at present there are four.
```

> ### Usecase(s)

Reviewer: ```suggestion
### Usecases
```

- Accidental deletion of data.
- Corruption of data due to application bugs.

> ### Precondition

Reviewer: ```suggestion
### Preconditions
```

> - There should be regular backup taken before point in time.

Reviewer: ```suggestion
- There should be a vitess backup taken before the desired point in time.
```

> - There should be binlogs available till the required point.

Reviewer: ```suggestion
- There should be continuous binlogs available from the backup time to the desired point in time.
```

### Usage

> #### Pre-setup

Reviewer: I think this heading can be removed.
```suggestion
```

> To use this feature, you need to have regular backup of vitess data and continuous binlogs. Once all this setup is done, you can checkout how to recover.

Reviewer: ```suggestion
To use this feature, you need a usable backup of vitess data and continuous binlogs. .
```

> Here is how you can take regular backup.

Reviewer: ```suggestion
Here is how you can create a backup.
```

`vtctlclient -server localhost:15999 Backup zone1-101`

Here `localhost:15999` is the address of vtctld grpc server. `zone1-101` is the tablet alias of a replica tablet in shard.

> To maintain continuous binlog, you need to have a binlog server pointing to the master. You can use [ripple](https://github.com/google/mysql-ripple). You need to take care of following things.

Reviewer: ```suggestion
To maintain continuous binlogs, you need to have a binlog server pointing to the master. You can use [ripple](https://github.com/google/mysql-ripple). You need to take care of the following.
```

 - You need to have a highly available binlog server setup, if it is down, we should be able to catch up before master rotates the binlog.
 - The binlog files should be safely kept at some reliable and recoverable location (i.e. AWS S3, remote file storage). 
 - In case of any reparenting or master failover, the binlog server should be able to switch to the new master.
 

>  Once above is done, you can proceed to send recover request. Here is how you can do a recovery.

Reviewer: ```suggestion
 Once above is done, you can proceed to send a recovery request. Here is how you can do a recovery.
```
